src/models/transferSettlement/facade.js

```diff
@@ -0,0 +1,305 @@
+        if (transferStateChangeId.length === 0 || !transferStateChangeId[0].transferStateChangeId || transferStateChangeId.length > 1) {
+          throw ErrorHandler.Factory.createInternalServerFSPIOPError(`Unable to find transfer with COMMITTED state for transferId : ${transferId}`)
+        }
+
+        const participantPositionRecords = await knex('participantPosition')
+          .select('participantPositionId', 'value', 'reservedValue')
+          .where('participantCurrencyId', recordsToInsert[0].participantCurrencyId)
+          .orWhere('participantCurrencyId', recordsToInsert[1].participantCurrencyId)
+          .transacting(trx)
+
+        if (participantPositionRecords.length !== 2) {
+          throw ErrorHandler.Factory.createInternalServerFSPIOPError(`Unable to find all participantPosition records for ParticipantCurrency: {${recordsToInsert[0].participantCurrencyId},${recordsToInsert[1].participantCurrencyId}}`)
+        }
+        const participantPositionChangeRecords = participantPositionRecords.map(participantPositionRecord => {
+          participantPositionRecord.transferStateChangeId = transferStateChangeId[0].transferStateChangeId
+          return participantPositionRecord
+        })
+
+        await knex('participantPositionChange')
+          .insert(participantPositionChangeRecords)
+          .transacting(trx)
+
+        if (doCommit) {
+          await trx.commit
+        }
+      } catch (err) {
+        if (doCommit) {
+          await trx.rollback
+        }
+        throw err
+      }
+    }
+    if (trx) {
+      return await trxFunction(trx, false)
+    } else {
+      return await knex.transaction(trxFunction)
+    }
+  } catch (err) {
+    throw ErrorHandler.Factory.reformatFSPIOPError(err)
+  }
+}
+
+async function insertLedgerEntries (ledgerEntries, transferId, trx = null) {
+  Logger.info(`Ledger entries: ${JSON.stringify(ledgerEntries)}`)
+  try {
+    const knex = await Db.getKnex()
+    const trxFunction = async (trx, doCommit = true) => {
+      try {
+        for (const ledgerEntry of ledgerEntries) {
+          Logger.info(`Inserting ledger entry: ${JSON.stringify(ledgerEntry)}`)
+          await insertLedgerEntry(ledgerEntry, transferId, trx)
+        }
+        if (doCommit) {
+          await trx.commit
+        }
+      } catch (err) {
+        if (doCommit) {
+          await trx.rollback
+        }
+        throw err
+      }
+    }
+    if (trx) {
+      return await trxFunction(trx, false)
+    } else {
+      return await knex.transaction(trxFunction)
+    }
+  } catch (err) {
+    throw ErrorHandler.Factory.reformatFSPIOPError(err)
+  }
+}
+
+async function updateTransferSettlement (transferId, status, trx = null) {
+  Logger.info(Utility.breadcrumb(location, { method: 'updateTransferSettlement' }))
+  try {
+    const knex = await Db.getKnex()
+    const trxFunction = async (trx, doCommit = true) => {
+      try {
+        // Insert TransferParticipant ledger entry type.
```

Comment:
The DB code below should be a separate model function/operation that can be isolated and tested appropriately in isolation.

Reply:
As discussed with @elnyry-sam-k: flag this issue for a future story